packages/protocol/src/mdns/MdnsSocket.ts

```diff
@@ -0,0 +1,184 @@
+
+            ...message,
+
+            answers: [],
+            additionalRecords: [],
+        };
+
+        // Note - for size calculations we assume queries are relatively small.  We only split answers across messages
+        let encodedChunkWithoutAnswers = DnsCodec.encode(chunk);
+        let chunkSize = encodedChunkWithoutAnswers.length;
+
+        // Add answers, splitting message as necessary
+        for (const answer of message.answers ?? []) {
+            const answerEncoded = DnsCodec.encodeRecord(answer);
+
+            if (chunkSize + answerEncoded.length > MAX_MDNS_MESSAGE_SIZE) {
+                if (chunk.answers.length === 0) {
+                    // The first answer is already too big, log at least a warning
+                    logger.warn(
+                        `MDNS message with ${Diagnostic.json(
+                            chunk.queries,
+                        )} is too big to fit into a single MDNS message. Send anyway, but please report!`,
+                    );
+                }
+
+                // New answer do not fit anymore, send out the message
+                await this.#send(chunk, intf, unicastDest);
+
+                // Reset the message, length counter and included answers to count for next message
+                if (chunk.queries.length) {
+                    chunk.queries.length = 0;
+                    encodedChunkWithoutAnswers = DnsCodec.encode(chunk);
+                }
+                chunk.answers.length = 0;
+                chunkSize = encodedChunkWithoutAnswers.length + answerEncoded.length;
+            } else {
+                chunkSize += answerEncoded.length;
+            }
+
+            chunk.answers.push(answerEncoded);
+        }
+
+        // Add "additional records"...  We include these but only if they fit
+        const additionalRecords = message.additionalRecords ?? [];
+        for (const additionalRecord of additionalRecords) {
+            const additionalRecordEncoded = DnsCodec.encodeRecord(additionalRecord);
+            chunkSize += additionalRecordEncoded.length;
+            if (chunkSize > MAX_MDNS_MESSAGE_SIZE) {
+                break;
+            }
+            chunk.additionalRecords.push(additionalRecordEncoded);
+        }
+
+        await this.#send(chunk, intf, unicastDest);
+    }
+
+    async #send(message: DnsMessagePartiallyPreEncoded, intf?: string, unicastDest?: string) {
+        await this.#socket.send(DnsCodec.encode(message), intf, unicastDest);
+    }
+
+    async close() {
+        this.#isClosed = true;
+        await this.#socket.close();
+        if (this.#handlers) {
+            await MatterAggregateError.allSettled(this.#handlers);
+        }
+    }
+
+    #handleMessage(bytes: Uint8Array, sourceIp: string, sourceIntf: string) {
+        // Ignore if closed
+        if (this.#isClosed) {
+            return;
+        }
+
+        // Skip unsupported source interface (not sure why this would happen, copied from old code)
```

Comment:
we could remove and see where it breaks. Ok with that. I also do not remember anymore

Reply:
OK will take out